src/components/VNepaliDatePicker.vue

```diff
@@ -1,7 +1,7 @@
 <template>
     <div class="wrapper">
         <input type="text" placeholder="yyyy-mm-dd" ref="elm" :value="dateValue" :class="inputClasses" :disabled="disabled"/>
-        <slot v-if="allowClear" :on-clear="onClear" name="clear-btn">
+        <slot v-if="allowClear || !disabled" :on-clear="onClear" name="clear-btn">
```

Comment:
@Sandip124 I think we need to use the && operator here. As it stands, the clear button will be shown even if the allowClear props is false.

<table>
<tr>
	<td> allowClear
	<td> disabled
	<td> clear button rendered?
<tr>
	<td> false
	<td> false
	<td> YES
<tr>
	<td> true
	<td> false
	<td> YES
<tr>
	<td> true
	<td> true
	<td> YES
<tr>
	<td> false
	<td> true
	<td> NO
</table>